core/src/main/java/org/hibernate/ogm/jpa/impl/OgmStoredProcedureQuery.java

```diff
@@ -0,0 +1,182 @@
+		entityManager().checkOpen( true );
+		parameterRegistrations.add( procedureCall.registerParameter( position, type, mode ) );
+		return this;
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public StoredProcedureQuery registerStoredProcedureParameter(String parameterName, Class type, ParameterMode mode) {
+		entityManager().checkOpen( true );
+		parameterRegistrations.add( procedureCall.registerParameter( parameterName, type, mode ) );
+		return this;
+	}
+
+	@Override
+	public StoredProcedureQueryImpl setParameter(int position, Object value) {
+		log.debugf( "set value %s for parameter index : %d", value, position );
+		checkOpen( true );
+
+		try {
+			findNoSQLParameterRegistration( position ).bindValue( value );
+		}
+		catch (QueryParameterException e) {
+			entityManager().markForRollbackOnly();
+			throw new IllegalArgumentException( e.getMessage(), e );
+		}
+		catch (HibernateException he) {
+			throw entityManager.convert( he );
+		}
+
+		return this;
+	}
+
+	private <X> ParameterRegistration<X> findNoSQLParameterRegistration(int parameterPosition) {
+		if ( parameterRegistrations != null ) {
+			for ( ParameterRegistration<?> param : parameterRegistrations ) {
+				if ( param.getPosition() == null ) {
+					continue;
+				}
+				if ( parameterPosition == param.getPosition() ) {
+					return (ParameterRegistration<X>) param;
+				}
+			}
+		}
+		throw new IllegalArgumentException( "Parameter with that position [" + parameterPosition + "] did not exist" );
+	}
+
+	@Override
+	public boolean execute() {
+		return super.execute();
+	}
+
+	@Override
+	public int getUpdateCount() {
+		return super.getUpdateCount();
+	}
+
+	@Override
+	public List getResultList() {
+		try {
+			final NoSQLProcedureOutputImpl rtn = (NoSQLProcedureOutputImpl) outputs().getCurrent();
+			log.debugf( "getSynchronizedQuerySpaces: %s", this.procedureCall.getSynchronizedQuerySpaces() );
+
+			return rtn.getResultList();
+		}
+		catch (HibernateException he) {
+			throw entityManager().convert( he );
+		}
+		catch (RuntimeException e) {
+			entityManager().markForRollbackOnly();
+			throw e;
+		}
+	}
+
+	@Override
+	public Object getSingleResult() {
+		try {
+			final NoSQLProcedureOutputImpl rtn = (NoSQLProcedureOutputImpl) outputs().getCurrent();
+			return rtn.getSingleResult();
+		}
+		catch (HibernateException he) {
+			throw entityManager().convert( he );
+		}
+		catch (RuntimeException e) {
+			entityManager().markForRollbackOnly();
+			throw e;
+		}
+	}
+
+	@Override
+	public int getMaxResults() {
+		// if have not ParameterMode.REF_CURSOR
+		boolean hasOutParameters = false;
+		for ( ParameterRegistration parameterRegistration : parameterRegistrations ) {
+			// parameterRegistration.getMode().equals( ParameterMode.REF_CURSOR )
+			if ( parameterRegistration.getMode().equals( ParameterMode.OUT ) ) {
+				hasOutParameters = true;
+
+			}
+		}
+		int result = 0;
+		if ( hasOutParameters ) {
+
+		}
```

Comment:
What happens in this case?

Reply:
If you agree I will thow unsupportedexception. Becase 'out' and 'in_out' parmeters not supported. I wrote it in documentation.

Reply:
Sounds good

Reply:
fixed